src/Symfony/Component/Form/ChoiceList/Factory/DefaultChoiceListFactory.php

```diff
@@ -81,6 +81,12 @@ public function createView(ChoiceListInterface $list, $preferredChoices = null,
                     $otherViews
                 );
             }
+
+            if ($groupByOrder && is_array($groupByOrder)) {
+                uksort($otherViews, function ($key1, $key2) use ($groupByOrder) {
```

Comment:
I'd like to see this support both callables and predefined arrays.  Something like this should work:

```php
if ($groupByOrder) {
    if (is_callable($groupByOrder)) {
        // TODO
    } elseif (is_array($groupByOrder)) {
        // TODO
    }
}
```